dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsPermissionExtractionTask.java

```diff
@@ -77,14 +78,17 @@ private void doRun(Writer output) throws IOException {
     conf.set("fs.defaultFS", "hdfs://" + clusterHost + ":" + port + "/");
 
     try (FileSystem fs = FileSystem.get(conf);
-        ScanContext scanCtx = new ScanContext(fs, output)) {
+        ScanContext scanCtx = new ScanContext(fs, output);
+        ExecutorManager execManager =
+            new ExecutorManager(
+                ExecutorManager.newExecutorServiceWithBackpressure(
+                    "hdfs-permission-extraction", poolSize))) {
```

Comment:
I don't understand the meaning of "a new, ExecutorService which we never shut down" - I see no shutdown() method in ExecutorManager. Is it not enough to get it await()-ed and don't post new tasks/jobs to its queue?

Reply:
Besides:
1. I see no `TaskRunContext.getExecutorService()`
2. Using `ForkJoinPool.commonPool()` is contradicting the requirement to use ExecutorManager, no?

Reply:
I think BigQuery uses its own ExecutorService because the one in TaskRunContext is sized for CPU-usage optimality, whereas the one in BQ is sized for the number of threads BQ APIs can cope with.

Reply:
To be concrete, The caller is responsible for calling MoreExecutors.shutdownAndWait() on the return value of ExecutorManager.newExecutorServiceWithBackpressure SEPARATELY from the await on the ExecutorManager, otherwise the threads persist. ExecutorManager is for managing tasks, not for managing threads.